### PR TITLE
Switching to Babel 6 compiler and adding testing infrastructure 

### PR DESCRIPTION
**Pull request by KamranAsif**
## Description
- Switched to babel 6 
- Removed .react file extensions
- Switched requires to import statments
- Removed erroneous trailing commas 
- Added testing framework with Mocha and Chai

## Motivation and Context
Lack of testing

## How Has This Been Tested?
It does itself :D 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: src/vendor_upstream/struct/PrefixIntervalTree.js [hunk 1]
```diff
@@ -55,18 +55,18 @@ class PrefixIntervalTree {
...
    * index of the first element in the heap. Always a power of 2.
    */
-  _half: number;
+  //_half: number;
 
   /**
    * Binary heap
    */
-  _heap: Array<number>;
+  //_heap: Array<number>;
```
Comment (KamranAsif) [inline src/vendor_upstream/struct/PrefixIntervalTree.js]: @wcjordan I think this used flow..Not sure if theres a better solution

File: examples/FilterExample.js [hunk 2]
```diff
@@ -55,7 +55,7 @@ class FilterExample extends React.Component {
   _onFilterChange(e) {
     if (!e.target.value) {
       this.setState({
-        filteredDataList: this._dataList,
+        filteredDataList: this._dataList
```
Comment (burakcan) [inline examples/FilterExample.js]: I think the trailing commas should be left since they will cause unnecessary git diffs in future.

Comment (KamranAsif) [inline examples/FilterExample.js]: @burakcan Unfortunately it fails the build

Comment (burakcan) [inline examples/FilterExample.js]: How?

Comment (KamranAsif) [inline examples/FilterExample.js]:
```
ERROR in ./src/FixedDataTable.js
Module build failed: SyntaxError: /Users/asif/Work/FDT/src/FixedDataTable.js: Unexpected token (1245:8)
  1243 |           cx('fixedDataTableLayout/horizontalScrollbar'),
  1244 |           cx('public/fixedDataTable/horizontalScrollbar'),
> 1245 |         )}
       |         ^
  1246 |         style={outerContainerStyle}>
  1247 |         <div style={innerContainerStyle}>
  1248 |           <Scrollbar
```

Comment (burakcan) [inline examples/FilterExample.js]: But this one (arguments) is different. Keeping trailing commas in objects and arrays wont break the build, i guess.

File: src/vendor_upstream/struct/PrefixIntervalTree.js (hunk 1 above)
Comment (wcjordan) [inline src/vendor_upstream/struct/PrefixIntervalTree.js]: Ya these look like flow annotations.  The code in constructor works fine with xs: Array<number> though... I wonder if the transforms only handle a subset of cases...

Should we leave the variables but document with jsdoc instead?

**Comment by wcjordan** [comment]
lgtm.  It's probably worth adding documentation or an example PR for adding a test.

**Comment by KamranAsif** [comment]
@burakcan I re-added all the commas and included babel-plugin-syntax-trailing-function-commas
I had to do a rebase and push forward because of the file renames

**Comment by KamranAsif** [comment]
@wcjordan Lets do document/example PR as a follow up

File: src/vendor_upstream/struct/PrefixIntervalTree.js (hunk 1 above)
Comment (KamranAsif) [inline src/vendor_upstream/struct/PrefixIntervalTree.js]: Done

**Comment by burakcan** [comment]
Just upgraded and working like a charm 💯

**Pull request merged**